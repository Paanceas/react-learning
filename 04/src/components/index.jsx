export * from './AddCategory';
export * from './GifGItem';
export * from './GifGrid';