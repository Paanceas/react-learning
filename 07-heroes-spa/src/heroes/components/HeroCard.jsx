import { Link } from "react-router-dom";


const CharacterByHero = ({alter_ego, characters}) => {
  if(alter_ego === characters) return(<></>);
  return (<p>{characters}</p>);
}

export const HeroCard = ({ hero }) => {

  const heroImg = `/assets/heroes/${hero.id}.jpg`;

  return (
    <div className="col ">
      <div className="card">
        <div className="row no-gutters">
          <div className="col-4">
            <img className="card-img" src={heroImg} alt={hero.superhero} />
          </div>
          <div className="col-8">
            <div className="card-body">
              <h5 className="card-title">{hero.superhero}</h5>
              <p className="card-text">{hero.alter_ego}</p>
              <CharacterByHero characters={hero.characters} alter_ego={hero.alter_ego} />
              <p className="card text">
                <small>{hero.first_appearance}</small>
              </p>
              <Link to={`/hero/${hero.id}`} >
                MÃ¡s...
              </Link>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
