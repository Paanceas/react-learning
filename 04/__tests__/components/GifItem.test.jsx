import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("pruebas en GifItem", () => {
  const props = {
    title: "Saitama",
    url: "https://google.com.co/img.png",
  };

  test("debe hacer match con el snapshot del componente", () => {
    const { container } = render(<GifItem {...props} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y el Alt indicado", () => {
    render(<GifItem {...props} />);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(props.url);
    expect(alt).toBe(props.title);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem {...props} />);
    expect(screen.getByText(props.title)).toBeTruthy();
  });
});
