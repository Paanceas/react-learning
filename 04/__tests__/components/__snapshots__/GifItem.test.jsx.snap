// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifItem debe hacer match con el snapshot del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://google.com.co/img.png"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
